### PR TITLE
add FilterRolesUsersConsumer

### PR DESCRIPTION
**Pull request by borovHokage**

### CODE REVIEW

File: src/RightsService.Broker/Consumers/FilterRolesUsersConsumer.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+      List<DbRole> roles = users.Select(u => u.Role).Distinct().ToList();
+
+      return roles.Select(r =>
+          new RoleFilteredData(
+            r.Id,
+            r.RoleLocalizations.FirstOrDefault()?.Name,
+            users.Where(u => u.RoleId == r.Id).Select(u => u.UserId).ToList()))
+        .ToList();
+    }
```
Comment (MariyaSin) [inline src/RightsService.Broker/Consumers/FilterRolesUsersConsumer.cs:30]: add empty line

File: src/RightsService.Broker/Consumers/FilterRolesUsersConsumer.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    {
+      List<DbUser> users = await _repository.GetUsersAsync(request.RolesIds);
+
+      List<DbRole> roles = users.Select(u => u.Role).Distinct().ToList();
+
+      return roles.Select(r =>
+          new RoleFilteredData(
+            r.Id,
+            r.RoleLocalizations.FirstOrDefault()?.Name,
+            users.Where(u => u.RoleId == r.Id).Select(u => u.UserId).ToList()))
```
Comment (MariyaSin) [inline src/RightsService.Broker/Consumers/FilterRolesUsersConsumer.cs:28]: do you need .Where(u => u.RoleId == r.Id)?

File: src/RightsService.Broker/Consumers/FilterRolesUsersConsumer.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+{
+  public class FilterRolesUsersConsumer : IConsumer<IFilterRolesRequest>
+  {
+    private readonly IUserRepository _repository;
+
+    private async Task<List<RoleFilteredData>> GetRolesDataAsync(IFilterRolesRequest request)
+    {
+      List<DbUser> users = await _repository.GetUsersAsync(request.RolesIds);
+
+      List<DbRole> roles = users.Select(u => u.Role).Distinct().ToList();
```
Comment (MariyaSin) [inline src/RightsService.Broker/Consumers/FilterRolesUsersConsumer.cs:22]: Why do you use distinct?

File: src/RightsService.Broker/Consumers/FilterRolesUsersConsumer.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      _repository = repository;
+    }
+
+    public async Task Consume(ConsumeContext<IFilterRolesRequest> context)
+    {
+      List<RoleFilteredData> rolesFilteredData = await GetRolesDataAsync(context.Message);
+
+      await context.RespondAsync<IOperationResult<IFilterRolesResponse>>(
+        OperationResultWrapper.CreateResponse((_) => IFilterRolesResponse.CreateObj(rolesFilteredData), context));
+
```
Comment (MariyaSin) [inline src/RightsService.Broker/Consumers/FilterRolesUsersConsumer.cs:43]: remove empty line

File: src/RightsService.Data/UserRepository.cs [hunk 5]
```diff
@@ -88,6 +88,14 @@ public async Task<DbUser> GetAsync(Guid userId)
       return await _provider.Users.Include(x => x.Rights).FirstOrDefaultAsync(x => x.UserId == userId);
     }
 
+    public async Task<List<DbUser>> GetUsersAsync(List<Guid> rolesIds)
```
Comment (MariyaSin) [inline src/RightsService.Data/UserRepository.cs:91]: you do not need use add the word "user" to method name in UserRepository

File: src/RightsService.Broker/Consumers/FilterRolesUsersConsumer.cs (hunk 1 above)
Comment (borovHokage) [reply src/RightsService.Broker/Consumers/FilterRolesUsersConsumer.cs:30]: fixed

File: src/RightsService.Broker/Consumers/FilterRolesUsersConsumer.cs (hunk 2 above)
Comment (borovHokage) [reply src/RightsService.Broker/Consumers/FilterRolesUsersConsumer.cs]: removed

File: src/RightsService.Data/UserRepository.cs (hunk 5 above)
Comment (borovHokage) [reply src/RightsService.Data/UserRepository.cs]: fixed

**Review by MariyaSin — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**